include/seqan/bed_io/read_bed.h

```diff
@@ -189,74 +189,93 @@ _readBedRecordNoData(BedRecord<Bed12> & record,
                      TForwardIter & iter,
                      CharString & buffer)
 {
-    IsNewline isNewline;
+    typedef AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> AssertNoNewline;
+
     // Read first three fields.
     _readBedRecordNoData(static_cast<BedRecord<Bed6 > &>(record), iter, buffer);
 
     // Read THICK BEGIN
     clear(buffer);
-    readUntil(buffer, iter, OrFunctor<IsTab, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
+    readUntil(buffer, iter, OrFunctor<IsTab, AssertNoNewline>());
     record.thickBegin = lexicalCast<int32_t>(buffer);
     skipOne(iter);
 
     // Read THICK END
     clear(buffer);
-    readUntil(buffer, iter, OrFunctor<IsTab, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
+    readUntil(buffer, iter, OrFunctor<IsTab, AssertNoNewline>());
     record.thickEnd = lexicalCast<int32_t>(buffer);
     skipOne(iter);
 
     // Read ITEM RGB
     clear(buffer);
-    readUntil(buffer, iter, OrFunctor<EqualsChar<','>, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
+    readUntil(buffer, iter, OrFunctor<IsTab, OrFunctor<EqualsChar<','>, AssertNoNewline> >());
     record.itemRgb.red = lexicalCast<int32_t>(buffer);
-    skipOne(iter);
+    if (value(iter) == ',')
+    {
+        skipOne(iter);
 
-    clear(buffer);
-    readUntil(buffer, iter, OrFunctor<EqualsChar<','>, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
-    record.itemRgb.green = lexicalCast<int32_t>(buffer);
-    skipOne(iter);
+        clear(buffer);
+        readUntil(buffer, iter, OrFunctor<EqualsChar<','>, AssertNoNewline>());
+        record.itemRgb.green = lexicalCast<int32_t>(buffer);
+        skipOne(iter);
 
-    clear(buffer);
-    readUntil(buffer, iter, OrFunctor<IsTab, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
-    record.itemRgb.blue = lexicalCast<int32_t>(buffer);
-    skipOne(iter);
+        clear(buffer);
+        readUntil(buffer, iter, OrFunctor<IsTab, AssertNoNewline>());
+        record.itemRgb.blue = lexicalCast<int32_t>(buffer);
+        skipOne(iter);
+    }
+    else if (value(iter) == '\t' && record.itemRgb.red == 0)
+    {
+        skipOne(iter);
+        record.itemRgb.green = 0;
+        record.itemRgb.blue = 0;
+    }
+    else
+    {
+        throw ParseError("While parsing field 9 (itemRGB) in BED12 file: "
+                         "value must be either 0 or a list of three integer values r,g,b.");
```

Comment:
Empty values are not allowed in optional fields?

Reply:
For consistency, yes. All other fields must not be empty as well (otherwise lexicalCast's would fail) and up to now the field was required to contain a value. However, I understand your point. Maybe _optional_ has to be understood as extension to BED6 format, and therefore it is an all-or-nothing optionality?

Reply:
Ok, I see. We will leave it as such.